README.md
![vite-template](https://realme-ten.vercel.app/api/v1/banner?colorA=1e3a8a&colorB=1d4ed8&textColor=60a5fa&title=vite-template&subtitle=NEO&desc=Play%20Vite%20with%20stitches%20and%20tailwindcss)

[![npm](https://img.shields.io/npm/v/@aiou/vite-template)](https://github.com/neo-hack/vite-template/tree/master) [![GitHub](https://img.shields.io/npm/l/@aiou/vite-template)](https://github.com/neo-hack/vite-template/tree/master) [![stackblitz](https://img.shields.io/badge/%E2%9A%A1%EF%B8%8Fstackblitz-online-blue)](https://stackblitz.com/github/neo-hack/vite-template/tree/master)

[Edit on StackBlitz ⚡️](https://stackblitz.com/github/neo-hack/vite-template/tree/master)


## features

- 💞 Play with `stitches` and `tailwindcss`
- 🗄️ Filed based route system with [vite-plugin-pages](https://github.com/hannoeru/vite-plugin-pages)
- 📑 Use `document.tsx` as html template with [vite-plugin-document](https://github.com/JiangWeixian/vite-plugin-document)
- 🔍 Inspect states of plugin with [vite-plugin-inspect](https://github.com/antfu/vite-plugin-inspect), enable via `process.env.VITE_INSPECT`. e.g. `VITE_INSPECT=true pnpm run dev`
- 🎨 Use icon with [css.gg](https://css.gg) via [tailwind-cssgg](https://github.com/JiangWeixian/tailwind-cssgg). e.g. `<i className="gg-zeit" />`

## development

- **Setup** - `pnpm i`
- **Build** - `pnpm build`

# 
<div align='right'>

*built with ❤️ by 😼*

</div>

CHANGELOG.md
# @aiou/vite-template

## 0.2.1

### Patch Changes

- a4114be: upgrade svgrs-plugins@0.1.1

## 0.2.0

### Minor Changes

- afb0a5a: vite@4

### Patch Changes

- ac78edd: add tailwind-cssgg
- afb0a5a: pin pnpm@7
- afb0a5a: upgrade eslint & automatic-jsx
- afb0a5a: add @svgr-rs/svgrs-plugin

## 0.1.2

### Patch Changes

- f86d1e9: add vite-plugin-inspect

## 0.1.1

### Patch Changes

- a29abaa: upgrade vite-plugin-document, fix not-working-with-react-router

## 0.1.0

### Minor Changes

- fd30657: add vite-plugin-document

### Patch Changes

- 6d58838: release mvp
- defa049: use vite3


src/App.css.ts
import { keyframes, style } from '@vanilla-extract/css'
import { sprinkles } from './styles/tailwindcss.css'

export const App = sprinkles({
  textAlign: 'center',
  height: 'screen',
  fontSize: '2xl',
})

const spin = keyframes({
  '0%': { transform: 'rotate(0deg)' },
  '100%': { transform: 'rotate(180deg)' },
})

export const AppLogo = style([
  sprinkles({
    height: '32',
    pointerEvents: 'none',
  }),
  {
    '@media': {
      '(prefers-reduced-motion: no-preference)': {
        animation: `${spin} infinite 20s linear`,
      },
    },
  },
])

export const AppHeader = sprinkles({
  background: 'slate-900',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  color: 'white',
  height: 'screen',
})

export const AppLink = sprinkles({
  color: 'cyan-400',
  fontSize: 'base',
})

export const button = sprinkles({
  fontSize: 'base',
})


src/pages/home/home.css.ts
import { keyframes, style } from '@vanilla-extract/css'
import { sprinkles } from '@/styles/tailwindcss.css'

export const App = sprinkles({
  textAlign: 'center',
  height: 'screen',
  fontSize: '2xl',
})

const spin = keyframes({
  '0%': { transform: 'rotate(0deg)' },
  '100%': { transform: 'rotate(180deg)' },
})

export const AppLogo = style([
  sprinkles({
    height: '32',
    pointerEvents: 'none',
  }),
  {
    '@media': {
      '(prefers-reduced-motion: no-preference)': {
        animation: `${spin} infinite 20s linear`,
      },
    },
  },
])

export const AppHeader = sprinkles({
  background: 'slate-900',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  color: 'white',
  height: 'screen',
})

export const AppLink = sprinkles({
  color: 'cyan-400',
  fontSize: 'base',
})

export const button = sprinkles({
  fontSize: 'base',
})


src/styles/tailwindcss.css.ts
import { defineProperties, createSprinkles } from '@vanilla-extract/sprinkles'
import { vars, responsive } from './vars.css'

const borderWidth = {
  '0': '0px',
  '1': '1px',
  '2': '2px',
  '4': '4px',
  '8': '8px',
}

// TODO: wild defined
const width = {
  xs: '20rem',
  sm: '24rem',
  md: '28rem',
  lg: '32rem',
  xl: '36rem',
  '2xl': '42rem',
  '3xl': '48rem',
  '4xl': '56rem',
  '5xl': '64rem',
  '6xl': '72rem',
  '7xl': '80rem',
}

// TODO:
// - container
const responsiveProperties = defineProperties({
  conditions: {
    mobile: {},
    tablet: { '@media': `screen and (min-width: ${responsive.md})` },
    desktop: { '@media': `screen and (min-width: ${responsive.lg})` },
    sm: { '@media': `screen and (min-width: ${responsive.sm})` },
    md: { '@media': `screen and (min-width: ${responsive.md})` },
    lg: { '@media': `screen and (min-width: ${responsive.lg})` },
    xl: { '@media': `screen and (min-width: ${responsive.xl})` },
    '2xl': { '@media': `screen and (min-width: ${responsive['2xl']})` },
  },
  defaultCondition: 'mobile',
  properties: {
    aspectRatio: ['auto', '1/1', '16/9'],
    display: ['none', 'flex'],
    flexDirection: ['row', 'column'],
    alignItems: ['stretch', 'flex-start', 'center', 'flex-end'],
    justifyContent: ['stretch', 'flex-start', 'center', 'flex-end'],
    gap: vars.space,
    paddingTop: vars.space,
    paddingBottom: vars.space,
    paddingLeft: vars.space,
    paddingRight: vars.space,
    marginTop: vars.space,
    marginBottom: vars.space,
    marginLeft: vars.space,
    marginRight: vars.space,
    boxShadow: vars.shadow,
    top: vars.space,
    left: vars.space,
    bottom: vars.space,
    right: vars.space,
    borderRadius: {
      none: '0px',
      sm: '0.125rem',
      DEFAULT: '0.25rem',
      md: '0.375rem',
      lg: '0.5rem',
      xl: '0.75rem',
      '2xl': '1rem',
      '3xl': '1.5rem',
      full: '9999px',
    },
    fontFamily: vars.fontFamily,
    fontSize: vars.fontSize,
    lineHeight: vars.lineHeight,
    textAlign: ['center'],
    pointerEvents: ['none'],
    fontWeight: {
      thin: 100,
      extralight: 200,
      light: 300,
      normal: 400,
      medium: 500,
      semibold: 600,
      bold: 700,
      extrabold: 800,
      black: 900,
    },
    borderWidth,
    borderTopWidth: borderWidth,
    borderLeftWidth: borderWidth,
    borderRightWidth: borderWidth,
    borderBottomWidth: borderWidth,
    listStyleType: ['none', 'disc', 'decimal'],
    height: {
      screen: '100vh',
      ...vars.space,
    },
    width: {
      screen: '100vw',
      ...vars.space,
    },
    minHeight: {
      0: '0',
      full: '100%',
      screen: '100vh',
    },
    maxWidth: {
      none: 'none',
      0: '0rem',
      ...width,
      full: '100%',
      min: 'min-content',
      max: 'max-content',
      prose: '65ch',
    },
    inset: [0],
    textDecorationLine: ['underline', 'overline', 'line-through', 'none'],
    textDecorationStyle: ['solid', 'double', 'dotted', 'dashed', 'wavy'],
    textUnderlineOffset: borderWidth,
    textDecorationThickness: borderWidth,
    list: {
      none: {
        'list-style-type': 'none',
      },
    },
    items: {
      center: {
        'align-items': 'center',
      },
    },
  },
  shorthands: {
    padding: ['paddingTop', 'paddingBottom', 'paddingLeft', 'paddingRight'],
    paddingX: ['paddingLeft', 'paddingRight'],
    paddingY: ['paddingTop', 'paddingBottom'],
    margin: ['marginTop', 'marginBottom', 'marginLeft', 'marginRight'],
    marginX: ['marginLeft', 'marginRight'],
    marginY: ['marginTop', 'marginBottom'],
    placeItems: ['alignItems', 'justifyContent'],
    typeSize: ['fontSize', 'lineHeight'],
    rounded: ['borderRadius'],
    underlineOffset: ['textUnderlineOffset'],
  },
})

const colorModeProperties = defineProperties({
  conditions: {
    lightMode: {},
    darkMode: { '@media': '(prefers-color-scheme: dark)' },
    hover: { selector: '&:hover' },
    focus: { selector: '&:focus' },
  },
  defaultCondition: 'lightMode',
  properties: {
    color: vars.color,
    borderColor: vars.color,
    background: vars.color,
    textDecorationColor: vars.color,
  },
})

export const sprinkles = createSprinkles(responsiveProperties, colorModeProperties)


src/styles/vars.css.ts
import { createGlobalTheme } from '@vanilla-extract/css'
import colors from 'tailwindcss/colors'

type TailwindColorStop =
  | 50
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900;

function tailwindColorMap<T extends keyof typeof colors>(
  name: T
): {
  // eslint-disable-next-line prettier/prettier
  [P in keyof typeof colors[T] as `${T}-${TailwindColorStop}`]: string;
} {
  // @ts-ignore: I promise the type is correct:
  return Object.fromEntries(
    Object.entries(colors[name]).map(([num, value]) => [
      `${name}-${num}`,
      value,
    ])
  );
}

export const responsive = {
  sm: "640px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  "2xl": "1536px",
};

export const vars = createGlobalTheme(':root', {
  shadow: {
    sm: '0 1px 2px 0 rgba(0, 0, 0, 0.05)',
    DEFAULT: '0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)',
    md: '0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)',
    lg: '0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)',
    xl: '0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)',
    '2xl': '0 25px 50px -12px rgba(0, 0, 0, 0.25)',
    inner: 'inset 0 2px 4px 0 rgba(0, 0, 0, 0.06)',
    none: 'none',
  },
  space: {
    auto: 'auto',
    px: '1px',
    "0": '0px',
    "0.5": '0.125rem',
    "1": '0.25rem',
    "1.5": '0.375rem',
    "2": '0.5rem',
    "2.5": '0.625rem',
    "3": '0.75rem',
    "3.5": '0.875rem',
    "4": '1rem',
    "5": '1.25rem',
    "6": '1.5rem',
    "7": '1.75rem',
    "8": '2rem',
    "9": '2.25rem',
    "10": '2.5rem',
    "11": '2.75rem',
    "12": '3rem',
    "14": '3.5rem',
    "16": '4rem',
    "20": '5rem',
    "24": '6rem',
    "28": '7rem',
    "32": '8rem',
    "36": '9rem',
    "40": '10rem',
    "44": '11rem',
    "48": '12rem',
    "52": '13rem',
    "56": '14rem',
    "60": '15rem',
    "64": '16rem',
    "72": '18rem',
    "80": '20rem',
    "96": '24rem',
  },
  color: {
    white: colors.white,
    black: colors.black,
    transparent: "transparent",
    ...tailwindColorMap("blue"),
    ...tailwindColorMap("cyan"),
    ...tailwindColorMap("gray"),
    ...tailwindColorMap("red"),
    ...tailwindColorMap("green"),
    ...tailwindColorMap("indigo"),
    ...tailwindColorMap("purple"),
    ...tailwindColorMap("sky"),
    ...tailwindColorMap("slate"),
  },
  fontFamily: {
    body:
      '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
    code: "source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace",
  },
  fontSize: {
    xs: '0.75rem',
    sm: '0.875rem',
    base: '1rem',
    lg: '1.125rem',
    xl: '1.25rem',
    '2xl': '1.5rem',
    '3xl': '1.875rem',
    '4xl': '2.25rem',
    '5xl': '3rem',
    '6xl': '3.75rem',
    '7xl': '4.5rem',
    '8xl': '6rem',
    '9xl': '8rem',
  },
  lineHeight: {
    xs: '1rem',
    sm: '1.25rem',
    base: '1.5rem',
    lg: '1.75rem',
    xl: '1.75rem',
    '2xl': '2rem',
    '3xl': '2.25rem',
    '4xl': '2.5rem',
    '5xl': '1',
    '6xl': '1',
    '7xl': '1',
    '8xl': '1',
    '9xl': '1',
  },
})
